File: tests/benchmarks/py/math.py
from math import floor, sqrt
from _harness import run


def bench_math(it):
    r = 1.0
    i = 0
    while i < it:
        r = abs(r - i)
        r = sqrt(r + 1.0)
        r = floor(r * 10.0) / 10.0
        i = i + 1
    return r


run(lambda: bench_math(500000))

File: tests/benchmarks/js/math.js
function benchMath(it){ let r=1.0,i=0; while(i<it){ r=Math.abs(r-i); r=Math.sqrt(r+1.0); r=Math.floor(r*10.0)/10.0; i=i+1; } return r; }
function compute(){ return benchMath(500000); }
for(let w=0;w<3;w++) compute();
let best=Infinity, chk;
for(let r=0;r<10;r++){ const t=performance.now(); chk=compute(); const ms=performance.now()-t; if(ms<best)best=ms; }
console.error(chk); console.log(best.toFixed(3));
